api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ResumoMensalCohortQueries.java

```diff
@@ -111,17 +111,87 @@ public CohortDefinition getPatientsWhoInitiatedPreTarvAtAfacilityDuringCurrentMo
     cd.addSearch(
         "A2I",
         map(sqlCohortDefinition, "startDate=${startDate},endDate=${endDate},location=${location}"));
-    cd.addSearch("A2III", map(getPatientsWithTransferFromOtherHF(), "locationList=${location}"));
     cd.addSearch(
         "A2II",
         map(
-            getPatientsWithFirstClinicalConsultationOnTheSameDateAsPreArtStartDate(),
-            "endDate=${endDate},location=${location}"));
+            getPatientsEnrolledInPreArtOrRegisteredInClinicalProcessOpening(),
+            "startDate=${startDate},endDate=${endDate},location=${location}"));
+    cd.addSearch("A2III", map(getPatientsWithTransferFromOtherHF(), "locationList=${location}"));
+    cd.addSearch("A2IV", map(getPatientsWithTransferFrom(), "locationList=${location}"));
+    cd.addSearch(
+        "A2V",
+        map(
+            getAllPatientsWithPreArtDateTransferredInFromWithBoundaries(),
+            "startDate=${startDate},endDate=${endDate},location=${location}"));
+    cd.addSearch(
+        "A2VI",
+        map(
+            getAllPatientsRegisteredAsTransferredInProgramEnrolmentWithBoundaries(),
+            "startDate=${startDate},endDate=${endDate},location=${location}"));
 
-    cd.setCompositionString("(A2I AND A2II) AND NOT A2III");
+    cd.setCompositionString("(A2I AND A2II) AND NOT (A2III AND A2IV AND A2V AND A2VI)");
```

Comment:
`A2III`, `A2IV` and `A2V` will probably need to be combined to a single query. It does not say in the spec but all obs need to be in the same encounter (BA can confirm this) and using separate queries does not guarantee that.

It should look similar to this query:
https://github.com/esaude/openmrs-module-eptsreports/blob/45eeb243a29bd4e180b92bd463c7b3def619dc02/api/src/main/java/org/openmrs/module/eptsreports/reporting/cohort/evaluator/EptsTransferredInCohortDefinitionEvaluator.java#L40-L94

Reply:
`EptsTransferredInCohortDefinition` implements the program clause. 

So `getAllPatientsRegisteredAsTransferredInProgramEnrolmentWithBoundaries` is not necessary anymore, it should be deleted.